### PR TITLE
{cae}[intel/2015a] GATE 6.2 (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: easybuild/easyconfigs/g/GATE/GATE-6.2-intel-2015a.eb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+sources = [('%%(namelower)s_v%s_tar_gz_15843.gz' % '_'.join(version.split('.')), "tar xfvz %s")]
+source_urls = ['http://www.opengatecollaboration.org/sites/opengatecollaboration.org/files/gate_release/2012/08/']
+
+patches = ['GATE-%(version)s_Makefile-prefix.patch']
+
+dependencies = [
+    ('Geant4', '9.5.p01'),
+    ('CLHEP', '2.1.1.0'),
+    ('ROOT', 'v5.34.01'),
+    ('CMake', '2.8.4'),
```
Comment (wpoely86) [inline easybuild/easyconfigs/g/GATE/GATE-6.2-intel-2015a.eb]: dep or builddep? And why such an old version of CMake?

Comment (boegel) [inline easybuild/easyconfigs/g/GATE/GATE-6.2-intel-2015a.eb]: should be build dep indeed, will fix

old version to keep this in sync with existing GATE 6.2 easyconfigs (this is just a different toolchain)

File: easybuild/easyconfigs/p/Python/Python-2.7.3-intel-2015a.eb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+source_urls = ['http://www.python.org/ftp/%(namelower)s/%(version)s/']
+sources = [SOURCE_TGZ]
+
+patches = ['python_libffi_int128_icc.patch']
+
+# python needs bzip2 to build the bz2 package
+dependencies = [
+    ('bzip2', '1.0.6'),
+    ('zlib', '1.2.7'),
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/Python/Python-2.7.3-intel-2015a.eb]: `intel-2015a` -> `zlib-1.2.8`

Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.3-intel-2015a.eb]: Python 2.7.3 -> zlib 1.2.7 :)

This is just a different toolchain of existing easyconfigs, I'm not touching software versions in here.

Comment (wpoely86) [inline easybuild/easyconfigs/p/Python/Python-2.7.3-intel-2015a.eb]: Don't agree here. We stick the zlib to the toolchain. 2 different zlibs for one toolchain is trouble.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 7 out of 11 (11 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/0e26865bdba79f65163b for a full test report.

File: easybuild/easyconfigs/p/Python/Python-2.7.3-intel-2015a.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.3-intel-2015a.eb]: We'll stick the zlib to the toolchain when it's included in the toolchain, yes.
zlib is also a dep for libxml2, which may in turn be a dep for other things that have zlib as a dep

I'm not going to start introducing conflicts on zlib versions by randomly bumping the zlib version somewhere.

For all I know, Python 2.7.3 may not even build with zlib 1.2.8 (I'm not going to find out).

Comment (wpoely86) [inline easybuild/easyconfigs/p/Python/Python-2.7.3-intel-2015a.eb]: Now, if you want to load something next to `GATE`, you also will have a zlib conflict.

Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.3-intel-2015a.eb]: That argument works both ways, imho.

Comment (wpoely86) [inline easybuild/easyconfigs/p/Python/Python-2.7.3-intel-2015a.eb]: yes, both `zlib` is a core dep. Practicality everything depends on it.

Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.3-intel-2015a.eb]: OK, I'll bite. Bumped to v1.2.8, retesting. I'll merge it when the test report is successful (previous test failed due to unneeded patch in Python easyconfig).

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 13 out of 16 (15 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/6856d9e17d5ccfe2396f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 11 out of 15 (15 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/c5fbd4845453a5120215 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 15 out of 15 (15 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/9023d4a605b6083dab51 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 15 out of 15 (15 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/a5cabe543829bb7b25db for a full test report.

**Comment by boegel** [comment]
OK, going in, thanks for the review @wpoely86!

**Pull request merged**